conformance/lang/expressions/xml-attribute-access-expr/xml_attribute_access_expr.balt

```diff
@@ -0,0 +1,626 @@
+            </book>
+            <planner id='13'>Daily Planner<kind id='14'>day</kind><pages id='15'>365</pages></planner>
+            <book id='20'>
+                <name>The Sign of Four</name>
+                <author><name>Arthur Conan Doyle</name></author>
+            </book>
+            <pen><kind>marker</kind><color id='21'>blue</color></pen>
+        </items>`;
+    io:println(x.<items>/<book>[0]?.id); // @output 10
+    io:println(x.<items>/<book>[0]?.idx == ()); // @output true
+    io:println(x.<items>/<book>[1]?.id); // @output 20
+    io:println(x.<items>/<book>[1]?.idx == ()); // @output true
+    io:println(x.<items>/<book>[0]/<author>?.id); // @output 11
+    io:println(x.<items>/<book>[0]/<author>?.idx == ()); // @output true
+    io:println(x.<items>/<book>[0]/<author>/<name>/<Date>?.id); // @output 12
+    io:println(x.<items>/<book>[0]/<author>/<name>/<Date>?.idx == ()); // @output true
+    io:println(x.<items>/<planner>/<pages>?.id); // @output 15
+    io:println(x.<items>/<planner>/<pages>?.idx == ()); // @output true
+    io:println(x.<items>/<pen>/<color>?.id); // @output 21
+    io:println(x.<items>/<pen>/<color>?.idx == ()); // @output true
+    io:println(x/*.<book>[0]/<author>?.id); // @output 11
+    io:println(x/*.<book>[0]/<author>?.idx == ()); // @output true
+    io:println(x/*.<pen>/<color>?.id); // @output 21
+    io:println(x/*.<pen>/<color>?.idx == ()); // @output true
+    io:println(x/<*>.<planner>/<kind>?.id); // @output 14
+    io:println(x/<*>.<planner>/<kind>?.idx == ()); // @output true
+    io:println(x/<*>.<pen>/<color>?.id); // @output 21
+    io:println(x/<*>.<pen>/<color>?.idx == ()); // @output true
+    io:println(x/**/<items>/<book>[0]/<author>/<name>/<Date>?.id); // @output 12
+    io:println(x/**/<items>/<book>[0]/<author>/<name>/<Date>?.idx == ()); // @output true
+    io:println(x/**/<items>/<book>[1]?.id); // @output 20
+    io:println(x/**/<items>/<book>[1]?.idx == ()); // @output true
+    io:println(x/**/<items>/<planner>/<kind>?.id); // @output 14
+    io:println(x/**/<items>/<planner>/<kind>?.idx == ()); // @output true
+    io:println(x/**/<items>/<pen>/<color>?.id); // @output 21
+    io:println(x/**/<items>/<pen>/<color>?.idx == ()); // @output true
+}
+
+Test-Case: output
+Description: Test xml-optional-attribute-access-expr when xml-attribute-name is an identifier.
+Fail-Issue: ballerina-platform/ballerina-lang#35146
+Labels: xml, xml-optional-attribute-access-expr, string, xml-step-expr
+
+function init() {
+    xml x = xml `<messages id="10">
+                    <note id="11">
+                        <to>Tove</to>
+                    </note>
+                    <note id="12">
+                        <to>Jani</to>
+                    </note>
+                </messages>`;
+    io:println(x/*.<note>[0]?.id); // @output 11
+    io:println(x/*.<note>[0]?.idx == ()); // @output true
+}
+
```

Comment:
We can probably add the same set of tests added for `xml-required-attribute-access-expr`, for `xml-optional-attribute-access-expr` too. 
The following crashes? 
```ballerina
xmlns "NSURL" as ns;

function init() {
    xml x = xml `<messages ns:id="10">
                    <note ns:id="11">
                        <to>Tove</to>
                    </note>
                    <note ns:id="12">
                        <to>Jani</to>
                    </note>
                </messages>`;
    io:println(x/*.<note>[0]?.ns:id); // @output 11
}
```

Reply:
https://github.com/ballerina-platform/ballerina-lang/issues/35146